### PR TITLE
Fix bug in MemorySnapshot

### PR DESCRIPTION
**Pull request by shargon**
# Description

Snapshot data is not taken into account during memory store reading

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

**Comment by shargon** [comment]
@Jim8y Any idea about how to fix `Seek` without lack of performance?

**Review by Jim8y — APPROVED** [review]

File: src/Neo/Persistence/MemorySnapshot.cs [hunk 1]
```diff
@@ -65,12 +65,22 @@ public IEnumerable<(byte[] Key, byte[] Value)> Seek(byte[] keyOrPrefix, SeekDire
...
         public byte[] TryGet(byte[] key)
         {
-            immutableData.TryGetValue(key, out byte[] value);
+            if (writeBatch.TryGetValue(key, out var value))
+            {
+                return value;
+            }
+
+            immutableData.TryGetValue(key, out value);
             return value?[..];
```
Comment (cschuchardt88) [inline src/Neo/Persistence/MemorySnapshot.cs:74]: This should return `null`? If not then change `writeBatch` to do the same.

**Review by cschuchardt88 — CHANGES_REQUESTED** [review]

File: src/Neo/Persistence/MemorySnapshot.cs (hunk 1 above)
Comment (shargon) [reply src/Neo/Persistence/MemorySnapshot.cs:74]: Yes, it returns null when it's not found

**Comment by shargon** [comment]
It seems that `LevelDb` and `RorcksRb` snapshot works exactly like that :S, I understand snapshot as a different thing...

File: src/Neo/Persistence/MemorySnapshot.cs (hunk 1 above)
Comment (cschuchardt88) [reply src/Neo/Persistence/MemorySnapshot.cs:74]: Do you see the difference? In the returns?

Comment (cschuchardt88) [reply src/Neo/Persistence/MemorySnapshot.cs:74]: Why are we creating a new pointer?

**Comment by cschuchardt88** [comment]
> It seems that `LevelDb` and `RorcksRb` snapshot works exactly like that :S, I understand snapshot as a different thing...

They use a bloom filter. Our memory store does not.

**Pull request closed without merging**